src/anyp/PortCfg.cc

```diff
@@ -101,3 +105,185 @@ AnyP::PortCfg::detailCodeContext(std::ostream &os) const
+    for (const auto &p: list) {
+        if (rawFlags.*p) {
+            debugs(3, DBG_CRITICAL, "FATAL: " << PortOptionStr(p) << " is unsupported on " << PortKindStr(rawFlags) <<  ' ' << detail);
+            self_destruct();
+        }
+    }
+}
+
+void
+AnyP::PortCfg::allowEither(const AnyP::TrafficModeFlags::List &list, const char *detail)
+{
+    const auto &rawFlags = flags.rawConfig();
+    int conflictedCount = 0;
+    SBuf conflicted;
+    for (const auto &p: list) {
+        if (rawFlags.*p) {
+            conflictedCount++;
+            if (!conflicted.isEmpty())
+                conflicted.append(',');
+            conflicted.append(PortOptionStr(p));
+        }
+    }
+    if (conflictedCount <= 1)
+        return;
+
+    debugs(3, DBG_CRITICAL, "FATAL: the combination of " << conflicted <<
+            " is unsupported on " << PortKindStr(rawFlags) << ' ' << detail);
+    self_destruct();
+}
+
+void
+AnyP::PortCfg::checkImplication(const AnyP::TrafficModeFlags::Pointer aFlag, const AnyP::TrafficModeFlags::List &list, const char *detail)
+{
+    const auto &rawFlags = flags.rawConfig();
+    if (!(rawFlags.*aFlag))
+        return;
+
+    for (const auto &p: list) {
+        if (rawFlags.*p)
+            return;
+    }
+
+    debugs(3, DBG_CRITICAL, "FATAL: " << PortOptionStr(aFlag) << " requires one of " << PortOptionStrList(list) <<
+            " on " << PortKindStr(rawFlags) << ' ' << detail);
+    self_destruct();
+}
+
+void
+AnyP::PortCfg::checkImplication(const AnyP::TrafficModeFlags::Pointer aFlag, const AnyP::TrafficModeFlags::Pointer otherFlag, const char *detail)
+{
+    const auto &rawFlags = flags.rawConfig();
+    if (!(rawFlags.*aFlag))
+        return;
+    if (!(rawFlags.*otherFlag)) {
+        debugs(3, DBG_CRITICAL, "FATAL: " << PortOptionStr(aFlag) << " requires " << PortOptionStr(otherFlag) <<
+                " on " << PortKindStr(rawFlags) << ' ' << detail);
+        self_destruct();
+    }
+}
+
+bool
+AnyP::PortCfg::hasAll(const AnyP::TrafficModeFlags::List &list)
+{
+    const auto &rawFlags = flags.rawConfig();
+    for (const auto &p: list) {
+        if (!(rawFlags.*p))
+            return false;
+    }
+    return true;
+}
+
+void
+AnyP::PortCfg::checkFlags()
+{
+    using Flags = AnyP::TrafficModeFlags;
+    const auto &rawFlags = flags.rawConfig();
+
+    switch (rawFlags.portKind) {
+
+    case Flags::httpPort: {
+        if (rawFlags.accelSurrogate)
+            rejectFlags({&TrafficModeFlags::natIntercept, &TrafficModeFlags::tproxyIntercept}, "accel");
+        else
+            allowEither({&TrafficModeFlags::natIntercept, &TrafficModeFlags::tproxyIntercept});
+    }
+    break;
+
+    case Flags::httpsPort: {
+        if (rawFlags.accelSurrogate) {
+            rejectFlags({&TrafficModeFlags::natIntercept, &TrafficModeFlags::tproxyIntercept,
+                    &TrafficModeFlags::proxySurrogateHttp, &TrafficModeFlags::tunnelSslBumping}, "accel");
+        } else {
```

Comment:
Same here: I think we can simplify by moving the mutual exclusivity check outside the `if` statement because it is universal or, in other words, because accelSurrogate is not more powerful than, say, natIntercept:

```suggestion
        rejectFlags({&TrafficModeFlags::natIntercept,
            &TrafficModeFlags::tproxyIntercept,
            &TrafficModeFlags::accelSurrogate,
            &TrafficModeFlags::proxySurrogateHttp,
            &TrafficModeFlags::tunnelSslBumping});
        if (!rawFlags.accelSurrogate) {
```

After this and other changes, you may be able to remove the `details` argument from rejectFlags().

Reply:
You probably meant 'alllowEither' here.
I figured out how to apply this suggestion and simplify other 'httpsPort' conditions, by means of checkFlagImplication() (renaming the old checkImplication()) and a new checkListImplication() (done at 6011846).